File: Aulas/Aula03/ExemploVetor2.go
package aula03

import (
	"fmt"
)

func main() {
	// Definindo o vetor de inteiros
	vetor := []int{5, 12, 7, 20, 15, 8, 3, 11, 6, 9}

	// Inicializando variáveis
	soma := 0
	maiorValor := vetor[0]
	menorValor := vetor[0]

	// Iterando sobre os elementos do vetor
	for _, valor := range vetor {
		soma += valor
		if valor > maiorValor {
			maiorValor = valor
		}
		if valor < menorValor {
			menorValor = valor
		}
	}

	// Calculando a média
	media := float64(soma) / float64(len(vetor))

	// Exibindo os resultados
	fmt.Printf("Média: %.2f\n", media)
	fmt.Printf("Maior valor: %d\n", maiorValor)
	fmt.Printf("Menor valor: %d\n", menorValor)
}

File: Aulas/Aula03/ExemploVetor2.java
package Aula03;

// Exemplo de Vetor 2 em Java - calculando a média dos elementos do vetor
// Mais o maior elemento do vetor e o menor elemento do vetor.

public class ExemploVetor2 {
    public static void main(String[] args) {
        int[] vetor = {5, 12, 7, 20, 15, 8, 3, 11, 6, 9};

        int soma = 0;
        int maiorValor = vetor[0];
        int menorValor = vetor[0];

        // Usando o loop for-each para percorrer o vetor
        for (int valor : vetor) {
            soma += valor;
            if (valor > maiorValor) {
                maiorValor = valor;
            }
            if (valor < menorValor) {
                menorValor = valor;
            }
        }

        // Calculando a média
        double media = (double) soma / vetor.length;

        // Exibindo os resultados
        System.out.println("Média: " + media);
        System.out.println("Maior valor: " + maiorValor);
        System.out.println("Menor valor: " + menorValor);
    }
}
